### PR TITLE
fixup pod list frontend

### PR DESCRIPTION
**Pull request by Raven24**
fix problem with empty hostname in collection,
add tooltips to action icons in the table

### CODE REVIEW

**Comment by jhass** [comment]
Shouldn't we also add a migration to drop them from the DB, as well as duplicates?

**Comment by Raven24** [comment]
sure, makes sense

**Comment by jhass** [comment]
Oh and after dropping duplicates, adding a uniqueness index on the host would make sense I guess.

**Comment by jhass** [comment]
The job failed for me

File: app/models/pod.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def offline_statuses
     end
 
     def check_all!
-      Pod.find_in_batches(batch_size: 20).each(&:test_connection!)
+      Pod.find_in_batches(batch_size: 20) { |batch| batch.each(&:test_connection!) }
```
Comment (diaspora-code-review) [inline app/models/pod.rb]: Space between { and | detected.

Comment (jhass) [inline app/models/pod.rb]: What's the idea here anyway? Worst case we load 1k records here, not too bad.

Comment (Raven24) [inline app/models/pod.rb]: I just thought it could save a few bytes of RAM here and there :)

Comment (jhass) [inline app/models/pod.rb]: Anyway, if you fix the style issue it's good to go I guess.

File: db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def down
+    remove_index :pods, :host
+  end
+
+  private
+
+  def remove_dupes
+    duplicates = Pod.group(:host).count.select { |k, v| v > 1 }.keys
```
Comment (diaspora-code-review) [inline db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb]: Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.

Comment (diaspora-code-review) [inline db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb]: Space between { and | detected.

File: spec/migrations/remove_duplicate_and_empty_pods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
+
+require "spec_helper"
+require Rails.root.join("db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb")
+
+describe RemoveDuplicateAndEmptyPods do
+  self.use_transactional_fixtures = false
+
+  before do
+    @previous_migration = 20150731123114
```
Comment (diaspora-code-review) [inline spec/migrations/remove_duplicate_and_empty_pods_spec.rb]: Separate every 3 digits in the integer portion of a number with underscores(_).

File: spec/migrations/remove_duplicate_and_empty_pods_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    @my_migration = 20150828132451
```
Comment (diaspora-code-review) [inline spec/migrations/remove_duplicate_and_empty_pods_spec.rb]: Separate every 3 digits in the integer portion of a number with underscores(_).

**Comment by Raven24** [comment]
added the migration, dunno if the spec for it is really of that much value...

**Comment by jhass** [comment]
Btw you can run pronto locally, just `bin/pronto run -c upstream/develop`

File: db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    duplicates.each do |pod|
+      pods = Pod.where(host: pod)
+      keep = pods.order(created_at: :asc).first
+
+      pods.where.not(id: keep.id).each(&:destroy)
+    end
```
Comment (jhass) [inline db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb]: I'd say construct a list of IDs and do a single `destroy_all`

File: db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def remove_empty_or_nil
+    Pod.where(host: [nil, ""]).each(&:destroy)
```
Comment (jhass) [inline db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb]: Use `destroy_all`

File: db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
+class RemoveDuplicateAndEmptyPods < ActiveRecord::Migration
+  def up
+    remove_dupes
+    remove_empty_or_nil
+
+    add_index :pods, :host, unique: true, length: 190 # 190*4 for utf8bin
```
Comment (jhass) [inline db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb]: You mean `utf8mb4`? ;)

**Comment by Raven24** [comment]
yeah, I always forget that -.-

**Comment by Raven24** [comment]
more fixes for the fix ;)

File: db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+  private
+
+  def remove_dupes
+    duplicates = Pod.group(:host).count.select {|_, v| v > 1 }.keys
+    duplicates.each do |pod|
+      pods = Pod.where(host: pod).order(created_at: :asc).pluck(:id)
+      pods.shift # keep the first one
+      Pod.where(id: pods).destroy_all
+    end
```
Comment (jhass) [inline db/migrate/20150828132451_remove_duplicate_and_empty_pods.rb]: I was more thinking of something like:

```rb
ids = duplicates.flat_map {|pod| Pod.where(host: pod).order(created_at: :asc).pluck(:id).tap(&:shift) }
Pod.where(id: ids).destroy_all
```

File: spec/factories.rb [hunk 9]
```diff
@@ -211,7 +211,7 @@ def r_str
   end
 
   factory :pod do
-    host "pod.example.com"
+    sequence(:host) { |n| "pod#{n}.example#{r_str}.com" }
```
Comment (diaspora-code-review) [inline spec/factories.rb]: Space between { and | detected.

**Comment by jhass** [comment]
Guess there's a missing (default) order to make a spec reliable: https://travis-ci.org/diaspora/diaspora/jobs/77922154#L510

**Comment by Raven24** [comment]
or a leftover in the db from the migration spec ... where I had to turn off transactions :)

**Comment by jhass** [comment]
Thanks!

**Pull request merged**